### PR TITLE
Update ginkgo to version v2

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
ginkgo has updated to v2 three years ago and we should update our test framework. I think the two most important features of ginkgo/v2 we could make use of is 
    1. nested tables to help create more succinct testing of ovn-kubernetes 
    2. the [FlakeAttempts()](https://onsi.github.io/ginkgo/MIGRATING_TO_V2#flakeattempts-decorator) decorator, which could help diagnose flaky tests and at minimum allow us to unblock development when a flaky test is identified  

#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
this PR should only effect the unit testing and not e2e, rerunning the e2e jobs

**Review by maiqueb — DISMISSED** [review]
Got up to the vendoring code ...

I'm trusting you used a tool to update vendor, because I didn't check file by file the vendored changes are good ...

File: go-controller/hack/test-go.sh [hunk 1]
```diff
@@ -61,7 +61,7 @@ function testrun {
     fi
     if grep -q "ginkgo" ."${path}"/*_test.go; then
 	    prefix=$(echo "${path}" | cut -c 2- | sed 's,/,_,g')
-        ginkgoargs="-ginkgo.v ${ginkgo_focus} -ginkgo.reportFile ${TEST_REPORT_DIR}/junit-${prefix}.xml"
+        ginkgoargs="-ginkgo.v ${ginkgo_focus} --ginkgo.junit-report ${TEST_REPORT_DIR}/junit-${prefix}.xml"
```
Comment (jcaamano) [inline go-controller/hack/test-go.sh:64]: I would just use single dash here as with the other ginkgo parameter

**Review by jcaamano — COMMENTED** [review]
minor comment and pending rebase to resolve conflicts

File: go-controller/hack/test-go.sh (hunk 1 above)
Comment (JacobTanenbaum) [reply go-controller/hack/test-go.sh]: done

**Review by jcaamano — APPROVED** [review]

**Pull request merged**